src/common/components/repository-row/index.js

```diff
@@ -73,11 +75,28 @@ class RepositoryRow extends Component {
   }
 
   onRegisterClick(repositoryUrl) {
-    const { snap } = this.props;
+    const { snap, dispatch, fullName } = this.props;
     const repository = parseGitHubRepoUrl(repositoryUrl);
     const { snapName } = this.state;
     const triggerBuilds = (!!snap.snapcraft_data);
-    this.props.dispatch(registerName(repository, snapName, triggerBuilds));
+    let clientValidationError = false;
+
+    if (/^-|-$/.test(snapName)) {
```

Comment:
Not sure if this check works properly, I was just able to enter name with hyphens in start and end and they passed validation (API call refused to register that name)